src/transformers/models/funnel/modeling_tf_funnel.py

```diff
@@ -298,19 +303,19 @@ def stride_pool_pos(self, pos_id, block_index):
         else:
             return pos_id[::2]
 
-    def relative_pos(self, pos, stride, pooled_pos=None, shift=1):
+    def relative_pos(self, pos, stride, pooled_pos=None, shift=1.0):
         """
         Build the relative positional vector between `pos` and `pooled_pos`.
         """
         if pooled_pos is None:
             pooled_pos = pos
 
         ref_point = pooled_pos[0] - pos[0]
-        num_remove = shift * pooled_pos.shape[0]
+        num_remove = shift * tf.cast(shape_list(pooled_pos)[0], dtype=ref_point.dtype)
```

Comment:
why do we use `shape_list` here and `.shape()` previously? I don't really understand what the standard should be here